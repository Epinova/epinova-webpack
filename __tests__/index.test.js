const config = require("../config");
const serializer = require("jest-serializer-path");

expect.addSnapshotSerializer(serializer);

test("errors", () => {
    expect(() => config(undefined)).toThrowError();
    expect(() => config("dist", {})).toThrowError();
});

test("configuration", () => {
    const path = "public";

    const env = "test";
    const argv = {
        mode: "development"
    };

    const c = config({ path }, (config, configEnv, configArgv) => {
        config.entry = "./Scripts/global/index.js";

        expect(configEnv).toEqual(env);
        expect(configArgv).toEqual(argv);

        return config;
    });

    const value = c(env, argv);

    expect(typeof c).toEqual("function");
    expect(typeof value).toEqual("object");
    expect(value.entry).toEqual("./Scripts/global/index.js");
    expect(value.devServer.publicPath.includes(path)).toEqual(true);
    expect(value.output.publicPath).toEqual(`/${path}/`);
});

test("development", () => {
    expect(
        config({ path: "dist" })("development", { mode: "development" })
    ).toMatchSnapshot();
});

test("production", () => {
    expect(
        config({ path: "dist" })("production", { mode: "production" })
    ).toMatchSnapshot();
});
