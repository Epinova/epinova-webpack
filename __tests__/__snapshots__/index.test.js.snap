// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`development 1`] = `
Object {
  "devServer": Object {
    "compress": true,
    "contentBase": "<PROJECT_ROOT>",
    "disableHostCheck": true,
    "headers": Object {
      "Access-Control-Allow-Origin": "*",
    },
    "host": "0.0.0.0",
    "port": 8080,
    "publicPath": "http://127.0.0.1:8080/dist/",
  },
  "devtool": "cheap-module-source-map",
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "test": /\\\\\\.js\\$/,
      },
      Object {
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "<PROJECT_ROOT>/node_modules/mini-css-extract-plugin/dist/loader.js",
            "options": Object {
              "hmr": true,
              "reloadAll": true,
            },
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "importLoaders": 1,
              "sourceMap": true,
              "url": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "sourceMap": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.s\\(c\\|a\\)ss\\$/,
        "use": Array [
          Object {
            "loader": "<PROJECT_ROOT>/node_modules/mini-css-extract-plugin/dist/loader.js",
            "options": Object {
              "hmr": true,
              "reloadAll": true,
            },
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "importLoaders": 2,
              "sourceMap": true,
              "url": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "sourceMap": true,
            },
          },
          Object {
            "loader": "sass-loader",
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "minimizer": Array [
      OptimizeCssAssetsWebpackPlugin {
        "deleteAssetsMap": Object {},
        "options": Object {
          "assetNameRegExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
          "assetProcessors": Array [
            Object {
              "phase": "compilation.optimize-chunk-assets",
              "processor": [Function],
              "regExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
            },
          ],
          "canPrint": undefined,
          "cssProcessor": [Function],
          "cssProcessorOptions": Object {
            "discardComments": Object {
              "removeAll": true,
            },
          },
          "cssProcessorPluginOptions": Object {},
        },
        "phaseAssetProcessors": Object {
          "compilation.optimize-assets": Array [],
          "compilation.optimize-chunk-assets": Array [
            Object {
              "phase": "compilation.optimize-chunk-assets",
              "processor": [Function],
              "regExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
            },
          ],
          "emit": Array [],
        },
        "pluginDescriptor": Object {
          "name": "OptimizeCssAssetsWebpackPlugin",
        },
      },
      TerserPlugin {
        "options": Object {
          "cache": false,
          "cacheKeys": [Function],
          "chunkFilter": [Function],
          "exclude": undefined,
          "extractComments": false,
          "include": undefined,
          "minify": undefined,
          "parallel": true,
          "sourceMap": false,
          "terserOptions": Object {
            "output": Object {
              "comments": false,
            },
          },
          "test": /\\\\\\.m\\?js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
          "warningsFilter": [Function],
        },
      },
    ],
    "splitChunks": Object {
      "chunks": "initial",
    },
  },
  "output": Object {
    "filename": "[name].js",
    "hashDigestLength": 8,
    "path": undefined,
    "publicPath": "/dist/",
  },
  "plugins": Array [
    ManifestPlugin {
      "opts": Object {
        "basePath": "",
        "fileName": "manifest.json",
        "filter": null,
        "generate": null,
        "map": null,
        "publicPath": null,
        "seed": null,
        "serialize": [Function],
        "sort": null,
        "transformExtensions": /\\^\\(gz\\|map\\)\\$/i,
        "writeToFileEmit": true,
      },
    },
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "[name].css",
        "filename": "[name].css",
        "ignoreOrder": false,
        "moduleFilename": [Function],
      },
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".wasm",
      ".mjs",
      ".js",
      ".json",
      ".vue",
      ".jsx",
    ],
  },
}
`;

exports[`production 1`] = `
Object {
  "devServer": Object {
    "compress": true,
    "contentBase": "<PROJECT_ROOT>",
    "disableHostCheck": true,
    "headers": Object {
      "Access-Control-Allow-Origin": "*",
    },
    "host": "0.0.0.0",
    "port": 8080,
    "publicPath": "http://127.0.0.1:8080/dist/",
  },
  "devtool": false,
  "module": Object {
    "rules": Array [
      Object {
        "exclude": /node_modules/,
        "loader": "babel-loader",
        "test": /\\\\\\.js\\$/,
      },
      Object {
        "test": /\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "<PROJECT_ROOT>/node_modules/mini-css-extract-plugin/dist/loader.js",
            "options": Object {
              "hmr": false,
              "reloadAll": true,
            },
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "importLoaders": 1,
              "sourceMap": true,
              "url": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "sourceMap": true,
            },
          },
        ],
      },
      Object {
        "test": /\\\\\\.s\\(c\\|a\\)ss\\$/,
        "use": Array [
          Object {
            "loader": "<PROJECT_ROOT>/node_modules/mini-css-extract-plugin/dist/loader.js",
            "options": Object {
              "hmr": false,
              "reloadAll": true,
            },
          },
          Object {
            "loader": "css-loader",
            "options": Object {
              "importLoaders": 2,
              "sourceMap": true,
              "url": false,
            },
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "sourceMap": true,
            },
          },
          Object {
            "loader": "sass-loader",
          },
        ],
      },
    ],
  },
  "optimization": Object {
    "minimizer": Array [
      OptimizeCssAssetsWebpackPlugin {
        "deleteAssetsMap": Object {},
        "options": Object {
          "assetNameRegExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
          "assetProcessors": Array [
            Object {
              "phase": "compilation.optimize-chunk-assets",
              "processor": [Function],
              "regExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
            },
          ],
          "canPrint": undefined,
          "cssProcessor": [Function],
          "cssProcessorOptions": Object {
            "discardComments": Object {
              "removeAll": true,
            },
          },
          "cssProcessorPluginOptions": Object {},
        },
        "phaseAssetProcessors": Object {
          "compilation.optimize-assets": Array [],
          "compilation.optimize-chunk-assets": Array [
            Object {
              "phase": "compilation.optimize-chunk-assets",
              "processor": [Function],
              "regExp": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
            },
          ],
          "emit": Array [],
        },
        "pluginDescriptor": Object {
          "name": "OptimizeCssAssetsWebpackPlugin",
        },
      },
      TerserPlugin {
        "options": Object {
          "cache": false,
          "cacheKeys": [Function],
          "chunkFilter": [Function],
          "exclude": undefined,
          "extractComments": false,
          "include": undefined,
          "minify": undefined,
          "parallel": true,
          "sourceMap": false,
          "terserOptions": Object {
            "output": Object {
              "comments": false,
            },
          },
          "test": /\\\\\\.m\\?js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
          "warningsFilter": [Function],
        },
      },
    ],
    "splitChunks": Object {
      "chunks": "initial",
    },
  },
  "output": Object {
    "filename": "[name].[contenthash].js",
    "hashDigestLength": 8,
    "path": undefined,
    "publicPath": "/dist/",
  },
  "plugins": Array [
    ManifestPlugin {
      "opts": Object {
        "basePath": "",
        "fileName": "manifest.json",
        "filter": null,
        "generate": null,
        "map": null,
        "publicPath": null,
        "seed": null,
        "serialize": [Function],
        "sort": null,
        "transformExtensions": /\\^\\(gz\\|map\\)\\$/i,
        "writeToFileEmit": true,
      },
    },
    MiniCssExtractPlugin {
      "options": Object {
        "chunkFilename": "[name].[contenthash].css",
        "filename": "[name].[contenthash].css",
        "ignoreOrder": false,
        "moduleFilename": [Function],
      },
    },
  ],
  "resolve": Object {
    "extensions": Array [
      ".wasm",
      ".mjs",
      ".js",
      ".json",
      ".vue",
      ".jsx",
    ],
  },
}
`;
